src/python/LASSO.py

```diff
@@ -0,0 +1,309 @@
+merge_tb = mt.merge(hm,how='inner',left_on='CCL', right_on='Cell Line')
+MET = merge_tb.iloc[:,1:226]
+GCP = merge_tb.iloc[:,227:269]
+
+GCP = np.nan_to_num(GCP, nan=0)
+MET = np.nan_to_num(MET, nan=0)
+
+Xtrain, Xtest, Ytrain, Ytest =train_test_split(GCP, MET, test_size=0.3, random_state=0)
+print(Xtest.shape)
+print(Ytest.shape)
+print(Xtrain.shape)
+print(Ytrain.shape[1])
+
+
+
+# ### Tuning the alpha
+
+# In[4]:
+
+
+
+alphas = [10**-6, 10**-5,10**-4,10**-3,10**-2,10**-1,1,2,3,4,5,6,7,8,9,10]
+
+mdl = list()
+error = list()
+for i,a in enumerate(alphas):
+    for j in range(Ytrain.shape[1]):
+        x=linear_model.Lasso(alpha=a).fit(Xtrain, Ytrain[:, j])
+        mdl.append(x)
+    ypred = mdl[i].predict(Xtest)
+    error.append(mean_squared_error(ypred,Ytest[:, j]))
+print(error)
+minmse_index = error.index(min(error))
+best_alpha = alphas[minmse_index]
+print(best_alpha)
+
+
+# ### Fit train sets in Lasso with best alpha to generate models
+
+# In[5]:
+
+
+GCP2MET_models = []
+for i in range(Ytrain.shape[1]):
+    mdl_G2M =linear_model.Lasso(alpha=best_alpha).fit(Xtrain, Ytrain[:, i])
+#     mdl_G2M =linear_model.Lasso(alpha=0.01).fit(Xtrain, Ytrain[:, i])
+    GCP2MET_models.append(mdl_G2M)
+print(GCP2MET_models)
+
+
+MET2GCP_models = []
+
+for j in range(Xtrain.shape[1]):
+    mdl_M2G = linear_model.Lasso(alpha=best_alpha).fit(Ytrain, Xtrain[:,j])
+    MET2GCP_models.append(mdl_M2G)
+print(len(MET2GCP_models))
+
+
+# ### To evaluate models and get correlation coefficient, pvalue, and mse value.
+
+# In[6]:
+
+
+def evaluate_models(models, Xtest, Ytest):
+    """
+    evaluate_models returns results from the model predictions, including the pearson
+    correlation coefficient, p-Values, and MSE.
+
+    :param models:         A list of scikit-learn model objects.
+    :param Xtest:          A numpy array or pandas dataframe containing validation set input data.
+    :param Ytest:          A numpy array or pandas dataframe containing validation set output data.
+    :return pred_resul:    A dictionary containing the final MSE, pValue, or rValue.
+    
+    """
+    
+    predictions = []
+    rValue = list()
+    pValue = list()
+    MSE = list()
+    for i in range(len(models)):
+        mdl = models[i]
+        Ypred = mdl.predict(Xtest)
+        predictions.append(Ypred)   
+
+        r, pvalue = pearsonr(Ypred, Ytest[:, i])
+        rValue.append(r)
+        pValue.append(pvalue)
+        
+        mse = mean_squared_error(Ypred, Ytest[:, i])
+        MSE.append(mse)
+
+    df_MSE = pd.DataFrame(MSE)
+    df_pValue = pd.DataFrame(pValue)
+    df_rValue = pd.DataFrame(rValue)
+    
+    return df_MSE, df_pValue, df_rValue
+
+
+# ## GCP_to_MET
+# ### perform 3-fold cross validation to find model that generate the smallest mse value.
+
+# In[ ]:
+
+
+mse1,p1,pearson1 = evaluate_models(GCP2MET_models, Xtest, Ytest)
+mse1.columns = ["GCP2MET_models"]
+print(mse1)
+
+
+# In[7]:
+
+
+kf = KFold(n_splits=3)
```

Comment:
So I had you run hold out previously just to get the concept down. In practice, you don't have to run hold out. You can just report the results from k-fold CV,